### PR TITLE
refactor: Amount formatter

### PR DESCRIPTION
**Pull request by wal33d006**

### CODE REVIEW

File: packages/cosmos_utils/lib/amount_formatter.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+    if (raw.length >= 13) {
+      return _getShortAmount(raw, 12, 'T'); // Trillion
+    }
+    if (raw.length >= 10) {
+      return _getShortAmount(raw, 9, 'B'); // Billion
+    }
+    return _getShortAmount(raw, 6, 'M'); // Million
+  }
+  return formatted;
```
Comment (wal33d006) [inline packages/cosmos_utils/lib/amount_formatter.dart:18]: @andrzejchm 
Came up with this idea of adding `M`, `B`, `T`, and so on, for formatting amount along with simple formatting. Do let me know if we can think of something better than this.

Comment (andrzejchm) [reply packages/cosmos_utils/lib/amount_formatter.dart:18]: how about we use the `NumberFormat.compact()` formatter?

https://stackoverflow.com/a/54691084/4838100

Comment (wal33d006) [reply packages/cosmos_utils/lib/amount_formatter.dart:18]: That's perfect for our use case!

Comment (wal33d006) [reply packages/cosmos_utils/lib/amount_formatter.dart]: Updated ✅

File: packages/cosmos_utils/lib/amount_formatter.dart [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import 'package:intl/intl.dart';
+
+String formatAmount(double amount) => NumberFormat.compactCurrency(symbol: '\$').format(amount);
```
Comment (andrzejchm) [inline packages/cosmos_utils/lib/amount_formatter.dart:3]: let's make it a bit more sophisticated:
```suggestion
String formatAmount(double amount, {String locale = "en_US", String symbol = '\$'}) => amount >= 10000
    ? NumberFormat.compactCurrency(
        symbol: symbol,
        locale: locale,
      ).format(amount)
    : NumberFormat.currency(
        locale: locale,
        symbol: symbol,
      ).format(amount);

```

File: packages/cosmos_utils/test/number_formatter_test.dart [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import 'package:cosmos_utils/amount_formatter.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('Amount formatter test case', () {
+    expect(formatAmount(37570000000), '\$37.6B');
+  });
```
Comment (andrzejchm) [inline packages/cosmos_utils/test/number_formatter_test.dart:7]:
```suggestion
    test('Amount formatter test case', () {
    expect(formatAmount(37570000000), '\$37.6B');
    expect(formatAmount(10270, locale: "pl", symbol: "PLN"), '10,3 tys. PLN');
    expect(formatAmount(1027.004921), '\$1,027.00');
    expect(formatAmount(10000), '\$10K');
    expect(formatAmount(9999), '\$9,999.00');
    expect(formatAmount(1027), '\$1,027.00');
  });
```

**Review by andrzejchm — CHANGES_REQUESTED** [review]
I've added few more improvements so that the function is a bit more configurable

File: packages/cosmos_utils/test/number_formatter_test.dart (hunk 3 above)
Comment (wal33d006) [reply packages/cosmos_utils/test/number_formatter_test.dart]: ✅

File: packages/cosmos_utils/lib/amount_formatter.dart (hunk 2 above)
Comment (wal33d006) [reply packages/cosmos_utils/lib/amount_formatter.dart]: ✅

**Review by andrzejchm — APPROVED** [review]

**Pull request merged**